pkg/systemd/bootInfo.jsx

```diff
@@ -0,0 +1,138 @@
+import React, { useEffect, useState } from "react";
+import cockpit from "cockpit";
+import {
+    Card,
+    CardBody,
+    CardTitle, CodeBlock, CodeBlockCode,
+    EmptyStateHeader,
+    EmptyStateVariant,
+    List,
+    ListItem,
+    ListVariant, Spinner
+} from "@patternfly/react-core";
+import "./bootInfo.scss";
+import { EmptyState, EmptyStateBody } from "@patternfly/react-core/dist/esm/components/EmptyState";
+
+const _ = cockpit.gettext;
+
+export function BootInfo() {
+    const [svg, setSvg] = useState(undefined);
+    const [summary, setSummary] = useState(null);
+
+    useEffect(() => {
+        let string = "";
+        cockpit.spawn(["systemd-analyze", "plot"])
+                .stream((svg_part) => {
+                    string += svg_part;
+                })
+                .then(() => {
+                    const doc = new DOMParser().parseFromString(string, "text/xml");
+                    doc.querySelector("rect.background")?.remove();
+                    doc.querySelector("text[y='30']")?.remove();
+                    const svgElem = doc.querySelector("svg");
+                    svgElem.style.scale = "0.5";
+                    const summaryElem = doc.querySelector("text[y='50']");
+                    // Grabs summary from SVG and puts it in the card body instead
+                    if (summaryElem !== null) {
+                        setSummary(summaryElem.textContent);
+                        summaryElem.remove();
+                    }
+                    const [plot, legend] = doc.querySelectorAll("g");
+                    legend.remove();
+                    [...plot.querySelectorAll("text.left"), ...plot.querySelectorAll("text.right")].forEach((text) => {
+                        const match = text.innerHTML.match(/^(?<service>.+\.[a-z._-]+)(\s+\((?<time>\d+(\.\d+)?)\w+\))?$/);
```

Comment:
In my version of systemd-analyze (Fedora 39) the output SVG doesn't contain `<service>` tags. The services look like
```xml
  <text class="left" x="264.191" y="2674.000">initrd-parse-etc.service (52ms)</text>
  <rect class="activating" x="264.614" y="2680.000" width="0.000" height="19.000" />
  <rect class="active" x="264.614" y="2680.000" width="84.109" height="19.000" />
```

Reply:
This code doesn't look for a `<service>` tag. Are you perhaps confusing the named capture group with an HTML tag?

Reply:
Ah right, I did. But this looks even more brittle then -- processing XML with line-based REs sounds very brittle.

Reply:
It doesn't process any XML.

I'll use your example from above to explain

Here's a text element that is processed.
```svg
<text class="left" x="264.191" y="2674.000">initrd-parse-etc.service (52ms)</text>
```
The code gets the inner HTML, which is `initrd-parse-etc.service (52ms)` (innerText would return undefined). Then parses that using the regex.

The 3 named regex capturing groups then contain the following values:
|Group|Value|
|-|-|
|service|`initrd-parse-etc.service`|
|time|`52`|
|time_unit|`ms`|

Note: I changed the regex to split time and the unit with the latest commit

It won't matter if other text elements are added by `systemd-analyze` later with the same class because it won't do anything unless the format matches.